packages/api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -253,15 +254,44 @@ export const validateCart = async (
   const {
     clients: { commerce },
     loaders: { skuLoader },
+    headers
   } = ctx
 
   // Step1: Get OrderForm from VTEX Commerce
   const orderForm = await getOrderForm(orderNumber, session, ctx)
 
+  
+  // Validate if Session's cookie exists
+  const cookieSession = getCookie('vtex_session', headers.cookie) 
+
+  if (cookieSession && enableOrderFormSync === true) {
+
+    const {namespaces} =  await commerce.getsessionorder()  
+    const orderFormIdSession = namespaces.checkout?.orderFormId?.value 
+
+    // In the case of divergence between session cookie and indexdb update the order form
+    if (orderNumber != orderFormIdSession && orderFormIdSession!= undefined){
+
+      const orderFormSession = await getOrderForm(orderFormIdSession, session, ctx)
+      const isStale = isOrderFormStale(orderFormSession)
+
+      if (isStale === true && orderNumber) {
```

Comment:
I'm confusing if we should use `orderNumber` or `orderFormIdSession` here

Reply:
Should be orderFormIdSession since the orderNumber won't be the same as the session